crates/ty_python_semantic/src/types/builder.rs

```diff
@@ -374,6 +375,44 @@ impl<'db> UnionBuilder<'db> {
                     self.elements.swap_remove(index);
                 }
             }
+            Type::EnumLiteral(enum_literal) => {
+                let enum_class = enum_literal.enum_class(self.db);
+                let metadata =
+                    enum_metadata(self.db, enum_class).expect("Class of enum literal is an enum");
+
+                let members_in_union = self
```

Comment:
nit: a union has members as well as an enum; I'd find this name slightly clearer:

```suggestion
                let enum_members_in_union = self
```